accepted/2023/vscode-commands.md

```diff
@@ -0,0 +1,221 @@
+
+A new command is listed named `NuGet: Remove a package...``.
+
+![Alt text](../../meta/resources/vscode-commands/removepackage.png)
+
+When a developer selects this command, they will be prompted to select which project they would like to remove a package from with a dropdown of all of their projects. If the user only has one project, they will not see this option, and will skip to the next step.
+
+Next, the user will be prompted in a search box to provide a search term for a respective package to be found within the packages they have installed in their current project.  Beneath the search bar, a list of all packages installed in the project will appear, and the developer will have the option to enter a search term to narrow down the list, or just select directly from the list provided.
+
+![Alt text](../../meta/resources/vscode-commands/removepackage-searchbox.png)
+
+![Alt text](../../meta/resources/vscode-commands/removepackage-searchbox2.png)
+
+#### Option B: Solution Explorer Entry Points
+
+When a user hovers over a package that they have installed in their solution explorer, they will see two small icons appear, one of which being a 'minus sign'.
+
+![Alt text](../../meta/resources/vscode-commands/removeupdate-icon.png)
+
+Additionally, if the developer right clicks on a specific package in the folder, they will see a menu of options for operations to perform on this specific package.  One of these options will be: "Remove NuGet package...".
+
+![Alt text](../../meta/resources/vscode-commands/removeupdate-rightclick.png)
+
+### Removing a Package - Operation Flow
+
+Once the user selects their desired package to be removed through one of the two entry points, they will be prompted to confirm their choice with a pop-up   . If the user confirms this by clicking 'Remove', the package will be removed from their project.  They will also have the option to cancel the operation at this point.
+
+![Alt text](../../meta/resources/vscode-commands/removepackage-confirm.png)
+
+### Updating a Package - Entry Points
+
+#### Option A: Command Palette Entry Point
+
+A new command is listed named `NuGet: Update a package...``.
+
+![Alt text](../../meta/resources/vscode-commands/updatepackage.png)
+
+When a developer selects this command, they will be prompted to select which project they would like to update a package in with a dropdown of all of their projects. If the user only has one project, they will not see this option, and will skip to the next step.
+
+Next, the user will be prompted in a search box to provide a search term for a respective package to be found within the packages they have installed in their current project.   Beneath the search bar, a list of all packages installed in the project will appear, and the developer will have the option to enter a search term to narrow down the list, or just select directly from the list provided.
+
+![Alt text](../../meta/resources/vscode-commands/removepackage-searchbox.png)
```

Comment:
Should the results include the package's current and highest available version?

For example, I might be want to upgrade my ASP.NET Core dependencies to the latest 8.x version but not 9.x due to breaking changes.

Reply:
I think that is already provided in the [flow](https://github.com/NuGet/Home/blob/4e3eda9490122a456135caee0705d467d3ab4882/accepted/2023/vscode-commands.md#updating-a-package---operation-flow)? Let me know if your idea differs!

As for upgrading a major version of .NET for compatibility reasons, we don't support that functionality yet although it is a popular ask.

Reply:
Ah gotcha, I didn't look at the screenshot closely enough

Reply:
For whatever reason I cannot close my own comment threads. Feel free to mark this as resolved! :)